atariari/methods/encoders.py

```diff
@@ -68,16 +68,28 @@ def _make_layer(self, in_channels, depth):
             ResidualBlock(depth, depth)
         )
 
-    def forward(self, inputs):
-        out = inputs
-        if self.downsample:
-            out = self.layer3(self.layer2(self.layer1(out)))
-        else:
-            out = self.layer4(self.layer3(self.layer2(self.layer1(out))))
-        return F.relu(self.final_linear(self.flatten(out)))
+    @property
+    def convolution_depth(self):
+        return self.depths[-1]
+
+    def forward(self, inputs, fmaps=False):
```

Comment:
I assume these are the changes you needed to make ImpalaCNN work?

Reply:
Yes we needed to pass a convolutional layer as 'f5' to the InfoNCE trainers.